src/board.rs

```diff
@@ -0,0 +1,65 @@
+#[cfg(test)]
+mod test {}
+
+struct PlayerState {
+    king: u64,
+    queens: u64,
+    bishops: u64,
+    knights: u64,
+    rooks: u64,
+    pawns: u64,
+    king_castle: bool,
+    queen_castle: bool
+}
+
+impl PlayerState {
+    fn is_valid(&self) -> bool {
+        let mut valid = self.king.count_ones() == 1;
+        let add_board: u128 = self.king as u128 + self.queens as u128 + self.bishops as u128 + self.knights as u128 + self.rooks as u128 + self.pawns as u128;
+        valid &= add_board == self.all_pieces() as u128;
+        valid
+    }
+    fn all_pieces(&self) -> u64 {
+        self.king | self.queens | self.bishops | self.knights | self.rooks | self.pawns
+    }
+}
+
+enum Player {
+    Black,
+    White
+}
+
+//En Passant Target representation:
+//  0bX0000000: active flag. if 1, there was no en passant on the previous turn, and all other bits are ignored.
+//  0b0X000000: player flag. 0 for white, 1 for black.
+//  0b00XXXXXX: square of valid en passant target. bitboard is obtained by shifting 1u64 by this value.
```

Comment:
we should use `///` for documentation, as this will show up in editors and gets pulled out by tools like `rustdoc` (see https://doc.rust-lang.org/rustdoc/what-is-rustdoc.html for info)